src/java.base/share/classes/jdk/internal/vm/CDSParallelPreProcessor.java

```diff
@@ -89,15 +78,21 @@ public static int preLoadAndProcess(String classListFile,
         // here.
         List<List<String>> subLists = getSortedAndSplitClassList(
                 Files.readAllLines(Paths.get(classListFile)), parallelism);
-        Thread[] workers = new Thread[parallelism];
+        FutureTask<Void>[] tasks = new FutureTask[parallelism];
```

Comment:
Use callable instead of runnable, because I want to catch any exceptions from work threads and propagates to VM and exit directly.

Reply:
SG, thanks!